### PR TITLE
♻️Single method for unicode base64Url(str)

### PR DESCRIPTION
**Pull request by calebcordry**
Also use it for the `$BASE64` Macro.

cc/ @zikas

### CODE REVIEW

**Comment by lannka** [comment]
please move the new methods to the existing base64.js 
also, tests needed

File: extensions/amp-analytics/0.1/base64.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import {base64ToUrlSafe} from '../../../src/utils/base64';
+
+/**
+ * Base64 encoding that can handle unicode strings in a byte-efficent manner. As
+ * recommended by MDN;
+ * https://developer.mozilla.org/en-US/docs/Web/API/WindowBase64/Base64_encoding_and_decoding
+ * @param {str} str
+ * @return {str}
+ */
+export function b64EncodeUnicode(str) {
```
Comment (jridgewell) [inline extensions/amp-analytics/0.1/base64.js]: This is basically `base64EncodeFromBytes`.

I think you having written this is really caused by how bad the APIs provided by `src/utils/base64.js` are (if you can't tell, I hate every one of the helpers in there). None of them should be concerned with "bytes" (`Uint8Array`), it should just be string input, string output. If you cleaned that up, I'd be super happy.

Comment (calebcordry) [reply extensions/amp-analytics/0.1/base64.js]: I actually just realized the same thing :). I think a function that wraps `utf8Encode` and `base64UrlEncodeFromBytes` is the way to go.

File: test/functional/utils/test-base64.js [hunk 2]
```diff
@@ -169,3 +170,14 @@ describe('base64EncodeFromBytes', () => {
         .to.equal('/+8=');
   });
 });
+
+describe('base64EncodeFromString', () => {
+  it('should handle unicode and non-unicode strings', () => {
+    expect(base64UrlEncodeFromString('')).to.equal('');
+    expect(base64UrlEncodeFromString('helloworld')).to
```
Comment (lannka) [inline test/functional/utils/test-base64.js]: also add cases like:
- `  `: spaces 
- ` hello word ` leading, ending spaces
- ` $#!@#$%^&*() ` url unsafe symbols

Comment (calebcordry) [reply test/functional/utils/test-base64.js]: Done.

**Comment by calebcordry** [comment]
closes #17282

**Pull request merged**